test/system/src/main/java/io/pravega/test/system/framework/DockerRemoteSequential.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.test.system.framework;
+
+import com.google.common.collect.ImmutableMap;
+import com.spotify.docker.client.DefaultDockerClient;
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.VersionCompare;
+import com.spotify.docker.client.exceptions.DockerException;
+import com.spotify.docker.client.messages.ContainerConfig;
+import com.spotify.docker.client.messages.ContainerCreation;
+import com.spotify.docker.client.messages.HostConfig;
+import com.spotify.docker.client.messages.PortBinding;
+import io.pravega.common.Exceptions;
+import io.pravega.common.concurrent.Futures;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Assert;
+import java.lang.reflect.Method;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import static io.pravega.test.system.framework.Utils.DOCKER_NETWORK;
+import static io.pravega.test.system.framework.Utils.getConfig;
+import static org.junit.Assert.assertFalse;
+
+@Slf4j
+public class DockerRemoteSequential implements TestExecutor {
```

Comment:
This is confusing. There is a class called `RemoteSequential` that extends `TestExecutor`. This is creating a `DockerRemoteSequential` class, which logical should extend `RemoteSequential`, but it does. The naming of classes isn't working well.

Reply:
The name `DockerRemoteSequential` is used as the tests are running via docker swarm and they are executed remotely in a sequential manner. As this is not using any method from `RemoteSequential` , the inheritance is not required. But,  all the files which are needed for 
 different ways of execution  like docker based/ marathon based/ local exec/ distributed will extend from  TestExecutor, as they are different types of execution.

If there is a problem with this naming convention, I will change it . Can you please suggest a better one ?

Reply:
Let's talk in generic terms. Say I want to implement two kinds of servers that have some common functionality. I'd do, for example, something like:

```
interface Server {
...
}

class BlueServer implements Server {
... 
}

class YellowServer implements Server {
...
}
```

Both of the implementation classes have `Server` as a common suffix and that indicates some commonality. In the concrete case here, if I have one class called  `RemoteSequential` and another `DockerRemoteSequential`, then it sounds like the latter extends the former, which is not true at all here. Consequently, you have a couple of choices to make this right:

1- If `DockerRemoteSequential` and `RemoteSequential` have anything in common, then you need to rename the latter to say `DCOSRemoteSequential` and have them both extend or implement some common base class or interface called `RemoteSequential`.
2- If they share nothing other than the `TestExecutor`, then you better rename `DockerRemoteSequential` to something else.

It all boils down to understanding how much the two classes have in common.

Reply:
ok. done,  renamed it to `DockerBasedTestExecutor` as it doesn't have anything in common with `RemoteSequential` .